### PR TITLE
[WIP] Implement hangaman excercise

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
Fixes: #745

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
@m-a-ge I don't know much about writing test. Can you please guide me 😅

File: exercises/hangman/hangman_test.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import unittest
+import hangman
+
+
+class HangmanTest(unittest.TestCase):
+    def test(self):
+        self.assertEqual("hangman", "hangman")
+
```
Comment (m-a-ge) [inline exercises/hangman/hangman_test.py]: More tests should be added

File: exercises/hangman/example.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import random
+
+
+def main():
+    while True:
```
Comment (m-a-ge) [inline exercises/hangman/example.py]: I don't like this infinite loop

File: exercises/hangman/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+# Hangman
```
Comment (m-a-ge) [inline exercises/hangman/README.md]: You can adopt `README.md` from here - https://github.com/exercism/problem-specifications/blob/master/exercises/hangman/description.md

**Comment by m-a-ge** [comment]
@vaibhavsingh97, it's a best practice to write tests before working on actual implementation. It gives an opportunity to analyze the problem better, find edge cases etc.
We use `unittest` for writing tests, docs can be found here - https://docs.python.org/3/library/unittest.html 
You can also take a look at tests from different exercises as examples. 

You can also adopt tests from other tracks

File: exercises/hangman/example.py (hunk 2 above)
Comment (vaibhavsingh97) [reply exercises/hangman/example.py]: @m-a-ge This loop is only if player wan't to play again

File: exercises/hangman/hangman_test.py (hunk 1 above)
Comment (N-Parsons) [reply exercises/hangman/hangman_test.py]: @vaibhavsingh97, I would suggest basing the tests on the ones from the [C# track](https://github.com/exercism/csharp/blob/master/exercises/hangman/HangmanTest.cs) since they have fairly good test coverage.

**Comment by N-Parsons** [comment]
The [problem-specifications//description.md](https://github.com/exercism/problem-specifications/blob/master/exercises/hangman/description.md) suggests that this should be a functional reactive programming  (FRP) exercise using an external library. Such libraries seem to exist for Python, but I know nothing about them and I don't really understand what FRP really is. Unless you know what this is and can explain and implement it, I'm going to suggest that we just modify the exercise to use a more traditional paradigm. @m-a-ge, what are your thoughts on this?

Your current implementation is going to be really hard to write tests for, because it relies on user interaction, which can't be automated through `unittest` (as far as I know). A better approach might be to implement a `Hangman` class with methods for setting/picking a word and guessing letters, and write the tests based on this approach. It might be a cool extension to include something in the README about getting people to use their new class to write an interactive game as you've done - we can't test it, but people might find it quite fun.

It would be good to put whatever new description (based loosely on the `problem-specifications` description) into a `.meta/description.md` file in this exercise folder so that your new description is persistent when the `configlet` is used to regenerate the READMEs.

**Comment by N-Parsons** [comment]
@vaibhavsingh97 

The [problem-specifications//description.md](https://github.com/exercism/problem-specifications/blob/master/exercises/hangman/description.md) suggests that this should be a functional reactive programming  (FRP) exercise using an external library. Such libraries seem to exist for Python, but I know nothing about them and I don't really understand what FRP really is. Unless you know what this is and can explain and implement it, I'm going to suggest that we just modify the exercise to use a more traditional paradigm. @m-a-ge, what are your thoughts on this?

Your current implementation is going to be really hard to write tests for, because it relies on user interaction, which can't be automated through `unittest` (as far as I know). A better approach might be to implement a `Hangman` class with methods for setting/picking a word and guessing letters, and write the tests based on this approach. It might be a cool extension to include something in the README about getting people to use their new class to write an interactive game as you've done - we can't test it, but people might find it quite fun.

It would be good to put whatever new description (based loosely on the `problem-specifications` description) into a `.meta/description.md` file in this exercise folder so that your new description is persistent when the `configlet` is used to regenerate the READMEs.

**Comment by m-a-ge** [comment]
Personally, I'm in favor of adding a new exercise even if it contradicts `problem-specification` in some minor way.
It'll allow users to work on it, and we can always rewrite the example solutions to meet certain standards later (when we have time) by putting it to our backlog. 
Such exercises shouldn't be included to `core` exercises though

@N-Parsons what do you think of such approach to the problems likes these?

**Comment by vaibhavsingh97** [comment]
@m-a-ge What should I do know?

**Comment by m-a-ge** [comment]
@vaibhavsingh97 we need more tests to be added

**Comment by N-Parsons** [comment]
@m-a-ge, sounds good to me.

**Comment by cmccandless** [comment]
@vaibhavsingh97 are you still working on this?

**Comment by vaibhavsingh97** [comment]
Yes, I forgot now will update PR
